gobblin-metrics/src/main/java/gobblin/metrics/influxdb/InfluxDBReporter.java

```diff
@@ -121,50 +125,133 @@ public static Builder builder(String name) {
     return new Builder(name);
   }
 
-  private Serie.Builder addGauge(String name, Gauge gauge, long timeStamp, Serie.Builder builder) {
-    return builder.values(timeStamp, name, gauge.getValue());
+  private void reportGauge(MetricContext context, String name, Gauge gauge, long timeStamp) {
+    this.influxDB.write(
```

Comment:
I don't know the real performance impact unless we use this in production with a large deployment so really cannot comment on this. But this will help a lot on query performance. If it turns out performing really bad, we can always change it back.